test/unit/controllers/ChoiceControllerSpec.scala

```diff
@@ -18,43 +18,48 @@ package unit.controllers
 
 import controllers.ChoiceController
 import forms.Choice
+import forms.Choice.AllowedChoiceValues.StandardDec
+import org.mockito.Mockito
+import org.scalatest.BeforeAndAfterEach
 import play.api.libs.json.{JsValue, Json}
 import play.api.test.Helpers._
 import unit.base.ControllerSpec
 import views.html.choice_page
 
-class ChoiceControllerSpec extends ControllerSpec {
+class ChoiceControllerSpec extends ControllerSpec with BeforeAndAfterEach {
   import ChoiceControllerSpec._
 
-  trait SetUp {
-    val choicePage = new choice_page(mainTemplate)
+  private val choicePage = new choice_page(mainTemplate)
 
-    val controller = new ChoiceController(
-      mockAuthAction,
-      mockCustomsCacheService,
-      mockExportsCacheService,
-      mockErrorHandler,
-      stubMessagesControllerComponents(),
-      choicePage
-    )(ec, minimalAppConfig)
+  private val controller = new ChoiceController(
```

Comment:
I just created those tests few days ago... way with trait was to have new controller for every test case.

If we want to use this BeforeAndAfterEach, it must be moved to the ControllerSpec.

There is no reason to extends every test with this trait.

Reply:
I want to have a decision about it as soon as possible, because I'm working around those tests

Reply:
Sure :) I am going to redo your work and then we can have a chat. 
I am taking my Wrestling spandex kit, so that we can work this one out like civilized folks.